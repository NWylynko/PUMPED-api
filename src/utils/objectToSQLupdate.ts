import { escape } from 'sqlstring';

const objectToSQLupdate = (fields: any) => {
  let sql = '';

  Object.keys(fields).forEach((partName, index, array) => {
    // if the current item is the length of the array then its the last item
    // the last item cant have a comma
    const comma = index === array.length - 1 ? '' : ',';

    sql += `${escape(partName)} = ${escape(fields[partName])}${comma} `;
  });

  return sql;
};

export default objectToSQLupdate;
