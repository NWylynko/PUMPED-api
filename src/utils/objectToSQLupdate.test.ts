import objectToSQLupdate from './objectToSQLupdate';

it('is empty', () => {
  const object = {
    test: 'value',
    nice: 'easy',
    easy: 'peazy lemon squeezy',
  };

  expect(objectToSQLupdate(object)).toMatchSnapshot();
});
