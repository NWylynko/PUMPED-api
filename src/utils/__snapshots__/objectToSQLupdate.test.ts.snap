// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is empty 1`] = `"'test' = 'value', 'nice' = 'easy', 'easy' = 'peazy lemon squeezy' "`;
